patches/server/0019-Add-TPS-From-Region.patch

```diff
@@ -0,0 +1,52 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
+Date: Mon, 3 Jun 2024 11:01:19 +0200
+Subject: [PATCH] Add TPS From Region
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index 273c37b63df3f14488586f9217c7b19a8f3d8ad5..c383fa047c386952e7904b3d645b8f55498e4f25 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -3114,6 +3114,41 @@ public final class CraftServer implements Server {
+         };
+     }
+ 
++    // Folia start
++    @Override
++    public double[] getTPS(org.bukkit.Location location) {
++        final int x = location.blockX() >> 4;
++        final int z = location.blockZ() >> 4;
++        final ServerLevel world = ((CraftWorld) location.getWorld()).getHandle();
++        return getTPSFromRegion(world, x, z);
++    }
++
++    @Override
++    public double[] getTPS(org.bukkit.Chunk chunk) {
++        final int x = chunk.getX();
++        final int z = chunk.getZ();
++        final ServerLevel world = ((CraftWorld) chunk.getWorld()).getHandle();
++        return getTPSFromRegion(world, x, z);
++    }
++
++    private double[] getTPSFromRegion(ServerLevel world, int x, int z) {
++        io.papermc.paper.threadedregions.ThreadedRegionizer.ThreadedRegion<io.papermc.paper.threadedregions.TickRegions.TickRegionData, io.papermc.paper.threadedregions.TickRegions.TickRegionSectionData>
++            region = world.regioniser.getRegionAtSynchronised(x, z);
++        if (region == null) {
++            return new double[]{ 20.0, 20.0, 20.0, 20.0, 20.0 };
++        } else {
++            io.papermc.paper.threadedregions.TickRegions.TickRegionData regionData = region.getData();
++            return new double[] {
++                regionData.getRegionSchedulingHandle().getTickReport5s(System.nanoTime()).tpsData().segmentAll().average(),
```

Comment:
I think there is already some internal storage of this, because typing tps will show it from the top regions. Would propose to consult that if possible.
![imagem](https://github.com/PaperMC/Folia/assets/25226624/62e6cb9d-eed7-4d97-9855-a42fdc37627e)

Reply:
> Wouldn't it be better to at least store the time in nano seconds? As i remember it's not the fastest method to be called so many times

Yes, calling it every time is not a good idea, so I preferred to store it in a variable (like /tps)